sidechain/consensus/common/src/error.rs

```diff
@@ -62,3 +65,9 @@ impl core::convert::From<codec::Error> for Error {
 		Self::Other(e.to_string().into())
 	}
 }
+
+impl From<sgx_status_t> for Error {
+	fn from(sgx_status: sgx_status_t) -> Self {
+		Self::Sgx(sgx_status)
+	}
+}
```

Comment:
Ah thanks for mentioning this case in the whole 'type parameter or associated type' discussion. It helped a lot making it clear for me 👍 ☺️